docs/user-guide/a2-03-wgsl-target-specific.md

```diff
@@ -0,0 +1,104 @@
+| SV_CullDistance<N> | *Not supported* |
+| SV_Coverage | `@builtin(sample_mask)` |
+| SV_CullPrimitive | *Not supported* |
+| SV_Depth | `@builtin(frag_depth)` |
+| SV_DepthGreaterEqual | *Not supported* |
+| SV_DepthLessEqual | *Not supported* |
+| SV_DispatchThreadID | `@builtin(global_invocation_id)` |
+| SV_DomainLocation | *Not supported* |
+| SV_GSInstanceID | *Not supported* |
+| SV_GroupID | `@builtin(workgroup_id)` |
+| SV_GroupIndex | `@builtin(local_invocation_index)` |
+| SV_GroupThreadID | `@builtin(local_invocation_id)` |
+| SV_InnerCoverage | *Not supported* |
+| SV_InsideTessFactor | *Not supported* |
+| SV_InstanceID | `@builtin(instance_index)` |
+| SV_IntersectionAttributes | *Not supported* |
+| SV_IsFrontFace | `@builtin(front_facing)` |
+| SV_OutputControlPointID | *Not supported* |
+| SV_PointSize<sup>note</sup> | *Not supported* |
+| SV_Position | `@builtin(position)` |
+| SV_PrimitiveID | *Not supported* |
+| SV_RenderTargetArrayIndex | *Not supported* |
+| SV_SampleIndex | `@builtin(sample_index)` |
+| SV_ShadingRate | *Not supported* |
+| SV_StartVertexLocation | *Not supported* |
+| SV_StartInstanceLocation | *Not suported* |
+| SV_StencilRef | *Not supported* |
+| SV_Target<N> | *Not supported* |
+| SV_TessFactor | *Not supported* |
+| SV_VertexID | `@builtin(vertex_index)` |
+| SV_ViewID | *Not supported* |
+| SV_ViewportArrayIndex | *Not supported* |
+
+*Note* that `SV_PointSize` is a unique keyword that HLSL doesn't have.
+
+
+Behavior of `discard`
+---------------------
+
+Slang's `discard` is translated to `discard` in WGSL, which behaves similar to `discard` in HLSL.
+
+
+Supported HLSL features when targeting WGSL
+-------------------------------------------
+
+Slang supports the following HLSL feature sets when targeting WGSL.
+
+| Feature set | Supported |
+| -- | -- |
+| ray tracing | No |
+| inline ray tracing | No |
+| mesh shader | No |
+| tessellation shader | No |
+| geometry shader | No |
+| wave intrinsics | No |
+| barriers | Yes |
+| atomics | Yes |
+
+Supported atomic types
+----------------------
+
+Please refer to [the table showing atomics support for each target](a2-target-specific-features.md#supported-atomic-types-for-each-target).
+
+
+ConstantBuffer, (RW/RasterizerOrdered)StructuredBuffer, (RW/RasterizerOrdered)ByteAddressBuffer
+-----------------------------------------------------------------------------------------------
+
+ConstantBuffer translates to the `uniform` address space with `read` access mode in WGSL.
+ByteAddressBuffer and RWByteAddressBuffer translate to `array<u32>` with the `read` and `read_write` access modes in WGSL, respectively.
+StructuredBuffer and RWStructuredBuffer with struct type T translate to `array<T>` with with the `read` and `read_write` access modes in WGSL, respectively.
+
+
+Specialization Constants
+------------------------
+
+Specialization constants in Slang map to override declarations in WGSL.
+
+
+Matrix type translation
```

Comment:
Mention that InterlockedXXX is not supported, and should use atomic<T> instead.

Mention all the EntryPoint parameter/output legalization as in Ellie's metal doc.

Mention parameter block are translated to bindings in its own space.

Mention everything that isn't supported in wgsl but allowed in slang and slang's strategy to legalize them. Check out bug fix/commit/issue history to find all the legalization work we've done.

Reply:
> Mention that InterlockedXXX is not supported, and should use atomic instead.
> 
> Mention all the EntryPoint parameter/output legalization as in Ellie's metal doc.
> 

Done.
I left out the entry about 'varying parameters' being passed in a single struct, since it doesn't seem to be true for WGSL.


> Mention parameter block are translated to bindings in its own space.
> 

Done, except I believe the terminology is 'bind group' rather than 'space' in WGSL.

> Mention everything that isn't supported in wgsl but allowed in slang and slang's strategy to legalize them. Check out bug fix/commit/issue history to find all the legalization work we've done.

`git log --oneline  -- source/slang/slang-ir-wgsl-legalize.*` gives the following list:

- 65de5452 Fix the WGSL error about semantic when a nested struct is inlined (#5506)
   - It sounds like it wouldn't change anything about the generated WGSL.
- 79056cd7 Legalize the Entry-point for WGSL (#5498)
    - I mentioned parameter and return type struct flattening.
- e22c0b72 Fix system semantics of SV_GroupIndex (#5496)
    - Already covered by the table mapping Slang system values to WGSL built-in attributes.
- b118451e Move switch statement bodies to their own lines (#5493)
    - Formatting change. No impact on generated WGSL.
- 7c2ff547 Various WGSL fixes. (#5490)
    - I could mention that we always add 'default' switch cases, but I don't see why it would be worth mentioning.
- f65d756b format
- 53684ed9 Fix handling of pointer logic in wgsl backend. (#5129)
    - I wrote about how we convert parameters to pointer types, including the special handling of pointers to struct fields.
- 170558c9 Initial WGSL support (#5006)